crc/services/workflow_service.py

```diff
@@ -862,6 +862,17 @@ def _process_properties(spiff_task, props):
                 app.logger.error(f'Failed to process task property {str(ue)}', exc_info=True)
         return props
 
+    @staticmethod
+    def workflow_id_from_spiff_task(spiff_task):
+        workflow = spiff_task.workflow
+        # Find the top level workflow
+        while WorkflowProcessor.WORKFLOW_ID_KEY not in workflow.data:
+            if workflow.outer_workflow != workflow:
+                workflow = workflow.outer_workflow
+            else:
+                break
+        return workflow.data[WorkflowProcessor.WORKFLOW_ID_KEY]
+
```

Comment:
I don't understand this black magic (workflow_id_from_spiff_task)
I understand that we're looking for `workflow_id` in the data, but don't understand why it would or would not be there.